README.md

```diff
@@ -36,44 +36,55 @@ LLM App reads a corpus of documents stored in S3 or locally, preprocesses them,
 
 ## Getting Started
 
+This section provides a general introduction on how to start using the app.
+
 ### Installation
 
-Clone the repository. Create a new environment and install the required packages:
+- **Clone the repository:** This is done with the `git clone` command followed by the URL of the repository:
+    ```bash
+    git clone https://github.com/pathwaycom/llm-app.git
+    ```
+
+- **Virtual Python Environment:** Create a new environment and install the required packages to isolate the dependencies of this project from your system's Python:
 
-```bash
-python -m venv pw-env && source pw-env/bin/activate
-cd llm-app/
-pip install --upgrade --extra-index-url https://packages.pathway.com/966431ef6ba -r requirements.txt
-```
+    ```bash
+    # Creates an env called pw-env and activates this environment.
+    python -m venv pw-env && source pw-env/bin/activate
+    # Navigate to the repository and install the packages in requirements.txt
+    cd llm_app/
+    pip install --upgrade --extra-index-url https://packages.pathway.com/966431ef6ba -r requirements.txt
+    ```
 ### Usage
 
-- Create an .env file in `llm_app/` directory and add the following environment variables:
-```bash
-PATHWAY_REST_CONNECTOR_HOST=127.0.0.1
-PATHWAY_REST_CONNECTOR_PORT=8080
-OPENAI_API_TOKEN=<Your Token>
-PATHWAY_CACHE_DIR=/tmp/cache
-```
-
-- Run the script using the command: 
-```bash 
-cd llm_app/
-python main.py --mode contextful
-```
-You can also run the app without the need for external APIs by using `local` mode.
-
-- Send REST queries (in a separate terminal window):
-```bash
-curl --data '{"user": "user", "query": "How to connect to Kafka in Pathway?"}' http://localhost:8080/ | jq
-
-curl --data '{"user": "user", "query": "How to use LLMs in Pathway?"}' http://localhost:8080/ | jq
-```
-
-- Test reactivity by adding a new file:
-```bash
-cp ./data/documents_extra.jsonl ./data/pathway-docs/
-curl --data '{"user": "user", "query": "How to use LLMs in Pathway?"}' http://localhost:8080/ | jq
-```
+1. **Environment Variables:** Create an .env file in `llm_app/` directory and add the following environment variables:
+    ```bash
+    PATHWAY_REST_CONNECTOR_HOST=127.0.0.1
+    PATHWAY_REST_CONNECTOR_PORT=8080
+    OPENAI_API_TOKEN=<Your Token>
+    PATHWAY_CACHE_DIR=/tmp/cache
+    ```
```

Comment:
To be foolproof, maybe add that: "You should replace "<Your Token>" put your OpenAI API Token. You can change values of other variables if you wish." or sth similar?